src/OrchardCore/OrchardCore.ContentManagement.Abstractions/ContentItemExtensions.cs

```diff
@@ -228,4 +228,29 @@ public static ContentItem Merge(this ContentItem contentItem, object properties,
 
         return contentItem;
     }
+
+    /// <summary>
+    /// Creates an in-memory clone of a content item.
+    /// </summary>
+    /// <param name="contentItem"></param>
+    /// <returns></returns>
+    public static ContentItem Clone(this ContentItem contentItem)
```

Comment:
Some consideration about perf here is that the parts will be materialized on every request now, not reused like they used to be.

Reply:
Maybe move `Clone()` in the class so that it can access fields ... and if one day we want to move it we won't have to do a breaking change.

Reply:
Yes, there is a small performance drawback here. We could of cause move the `Clone()` method into the `ContentItem` and let it also clone the materialized elements, but those are also not thread-safe and are we sure this will not lead to another problem later on?

Note that we can always create an instance method later and let the extension method call that one. No breaking change required and for now it is consistent with all the other methods.